### PR TITLE
Flux blog post tweaks

### PR DESCRIPTION
**Pull request by MikeInnes**
Mostly just trying to address some of the feedback from ML folks, but also generally trying to cut down the length in a few places (e.g. for details of solver options we can point to the manual).

@ChrisRackauckas

### CODE REVIEW

File: blog/_posts/2019-01-18-fluxdiffeq.md [hunk 1]
```diff
@@ -196,42 +194,6 @@ using Plots
 plot(sol)
 ```
 
-![LV Solution Plot](https://user-images.githubusercontent.com/1814174/51388169-9a07f300-1af6-11e9-8c6c-83c41e81d11c.png)
```
Comment (ChrisRackauckas) [inline blog/_posts/2019-01-18-fluxdiffeq.md]: Keep the plot IMO

File: blog/_posts/2019-01-18-fluxdiffeq.md [hunk 2]
```diff
@@ -196,42 +194,6 @@ using Plots
...
-this equation is to use the 5th order Runge-Kutta method `Tsit5()` (related to the `dopri` method,
-or `ode45` in MATLAB). We can also say that we want this solved with a relative error
-tolerance of `1e-8`, and that we want the solution saved at every `0.1` time points. This looks like:
-
-```julia
-sol = solve(prob,Tsit5(),saveat=0.1,reltol=1e-8)
-```
-
-There are a lot more arguments you can pass in to customise most aspects of the ODE solver,
-as
```
Comment (ChrisRackauckas) [inline blog/_posts/2019-01-18-fluxdiffeq.md]: should we at least link to the docs for discussion about the arguments, and explain the ones we use?

File: blog/_posts/2019-01-18-fluxdiffeq.md [hunk 3]
```diff
@@ -247,18 +209,26 @@ as `θ` in associated literature). The utility of this will be seen later.
...
-vector `p` (of size `n`) and spits out a new vector (of size `m`). You can do this
-manually using the
-[`remake` function in DifferentialEquations.jl](http://docs.juliadiffeq.org/latest/basics/problem.html#Modification-of-problem-types-1),
-or you can use the Flux.jl layers defined in DiffEqFlux.jl. Let's give those a spin.
-
-The most basic differential equation layer is `diffeq_rd`. The function is of
-the form `diffeq_rd(p,reduction,prob,args...;kwargs...)`. The way to understand this
-function is that it does the following. It takes in the parameter vector `p`,
+To understand embedding an ODE into a neural network, let's look at what a
+neural network layer actually is. A layer is really just a *differentiable
```
Comment (ChrisRackauckas) [inline blog/_posts/2019-01-18-fluxdiffeq.md]: well, whether differentiability is really necessary is debatable. We can keep this in though

File: blog/_posts/2019-01-18-fluxdiffeq.md [hunk 4]
```diff
@@ -247,18 +209,26 @@ as `θ` in associated literature). The utility of this will be seen later.
...
+function* which takes in a vector of size `n` and spits out a new vector of size
+`m`. That's it! Layers have traditionally been simple functions like matrix
+multiply, but in the spirit of [differentiable
+programming](https://julialang.org/blog/2018/12/ml-language-compiler) people are
+increasingly experimenting with much more complex functions, such as ray tracers and
+physics engines.
+
+Turns out that differential equations solvers fit this framework, too: A solve
```
Comment (ChrisRackauckas) [inline blog/_posts/2019-01-18-fluxdiffeq.md]:
```suggestion
Turns out that differential equations solvers fit this framework, too: a solve
```

File: blog/_posts/2019-01-18-fluxdiffeq.md [hunk 5]
```diff
@@ -247,18 +209,26 @@ as `θ` in associated literature). The utility of this will be seen later.
...
+take in some vector `p` (which might include parameters like the initial
```
Comment (ChrisRackauckas) [inline blog/_posts/2019-01-18-fluxdiffeq.md]:
```suggestion
takes in some vector `p` (which might include parameters like the initial
```

File: blog/_posts/2019-01-18-fluxdiffeq.md [hunk 6]
```diff
@@ -247,18 +209,26 @@ as `θ` in associated literature). The utility of this will be seen later.
...
+starting point), and outputs some new vector, the solution. Moreover it's
+differentiable, which means we can put it straight into a larger differentiable
```
Comment (ChrisRackauckas) [inline blog/_posts/2019-01-18-fluxdiffeq.md]: not necessarily. The examples here, yes.

Comment (ChrisRackauckas) [reply blog/_posts/2019-01-18-fluxdiffeq.md]: Again, we can keep this as a simplification

File: blog/_posts/2019-01-18-fluxdiffeq.md (hunk 1 above)
Comment (MikeInnes) [reply blog/_posts/2019-01-18-fluxdiffeq.md]: Whoops, yes

File: blog/_posts/2019-01-18-fluxdiffeq.md (hunk 6 above)
Comment (MikeInnes) [reply blog/_posts/2019-01-18-fluxdiffeq.md]: A later part of the post says:
> The core technical challenge: backpropagation through differential equation solvers. Let's end by explaining the technical issue that needed a solution to make this
all possible. The core to any neural network framework is the ability to
backpropagate derivatives in order to calculate the gradient of the loss function
with respect to the network's parameters. Thus if we stick an ODE solver as a
layer in a neural network, we need to backpropagate through it.

that seems inconsistent with "differentiability isn't necessary". What's your alternative way of combining neural nets and diffeqs?

Comment (ChrisRackauckas) [reply blog/_posts/2019-01-18-fluxdiffeq.md]: SDE solutions aren't differentiable because they are only Holder continuous with coefficient < 1/2 almost surely, but they train just fine. You don't need differentiability to have an idea of Taylor expansions and weak derivatives. This is a discussion for another time though.

Comment (ChrisRackauckas) [reply blog/_posts/2019-01-18-fluxdiffeq.md]: Also, a lot of the problems we are solving, even here, are only differentiable a.e.

Comment (MikeInnes) [reply blog/_posts/2019-01-18-fluxdiffeq.md]: I mean, OK, but that's clearly not how I'm using the word here. We can perhaps clarify in a footnote that we mean differentiability in the AD/ML sense, e.g. "weak piecewise differentiable" or whatever you think is best.

**Comment by ChrisRackauckas** [comment]
Your capitalizations after colons must be a British thing. I've never seen that before, but apparently it can be okay: https://writingcenter.unc.edu/tips-and-tools/semi-colons-colons-and-dashes/

**Comment by MikeInnes** [comment]
Just be glad I'm not removing all your `z`s :)
